gammapy/datasets/map.py

```diff
@@ -1114,13 +1114,20 @@ def plot_residuals(
         return ax_spatial, ax_spectral
 
     def stat_sum(self):
-        """Total statistic function value given the current model parameters."""
+        """Total statistic function value given the current model parameters and priors."""
+        prior_stat_sum = 0.0
+        if self.models is not None:
+            prior_stat_sum = self.models.parameters.prior_stat_sum()     
```

Comment:
I just wanted to repeat the point I made in the dev call I while ago:

```
self.models.parameters.prior_stat_sum() 
```
Only works for prior that are on individual parameters. This means we will not be able to handle e.g. prior with off-diagonal elements in the covariance matrix. As it is e.g. need for the unfolding use-case. The `Models.parameters`
 attributed is re-created on access and thus cannot hold any state. Thus I I think it is important to move the prior stat sum to:
 
```
Models.prior_stat_sum()
```

Reply:
However this could happen in a follow up PR, but it should happen before we advertise the new API.

Reply:
Yes, I agree this is a preliminary version to include only one-dimensional priors and should not be advertised yet. Only the drafted follow-up PR with multi-dimensional priors, the `PriorStatistics` and the attachment of the priors to the models should be.